src/Security.js

```diff
@@ -12,15 +12,24 @@
 
 import React, { useState, useEffect } from 'react';
 import { useHistory } from 'react-router-dom';
-import { toRelativeUrl, AuthSdkError } from '@okta/okta-auth-js';
+import { toRelativeUrl, DEFAULT_AUTH_STATE, AuthSdkError } from '@okta/okta-auth-js';
 import OktaContext from './OktaContext';
 import OktaError from './OktaError';
 
 const Security = ({ oktaAuth, onAuthRequired, children }) => { 
   const history = useHistory();
-  const [authState, setAuthState] = useState(oktaAuth.authStateManager.getAuthState());
+  const [authState, setAuthState] = useState(() => {
+    if (!oktaAuth) {
```

Comment:
Wow are we actually allowing it to work without `okta-auth-js` What is the concept? That`<Security>` can be created first, and then later re-rendered with an oktaAuth instance?

Reply:
it's a workaround to make the `<Security>` render error when `oktaAuth` is missing. The linter does not allow conditional hooks (render error before the hook)

https://github.com/okta/okta-react/pull/8/files/5c3aee19c504c57e3203ca802079d560e0e6b20c#diff-b511f09482238b22139073ef766c2326566a2d86519cc15b992399092c344cddR56

Reply:
Worthy of a comment then, methinks